pkg/chartutil/requirements.go

```diff
@@ -266,3 +268,127 @@ func ProcessRequirementsEnabled(c *chart.Chart, v *chart.Config) error {
 
 	return nil
 }
+
+// pathToMap creates a nested map given a YAML path in dot notation.
+func pathToMap(path string, data map[string]interface{}) map[string]interface{} {
```

Comment:
- [ ] I still find this function's algorithm hard to follow, but I believe you that it works.